test/common/tap-cdp

```diff
@@ -66,13 +67,18 @@ if not cdp.get_browser_path():
     print('1..0 # skip web browser not found')
     sys.exit(0)
 
+# pass the address through a separate fd, so that we can see g_debug() messages (which go to stdout)
+(addr_r, addr_w) = os.pipe()
+env = os.environ.copy()
+env["TEST_SERVER_ADDRESS_FD"] = str(addr_w)
+
 server = subprocess.Popen(opts.server,
-                          stdout=subprocess.PIPE,
-                          stdin=subprocess.PIPE,
-                          universal_newlines=True,
-                          close_fds=True)
+                          stdin=subprocess.DEVNULL,
+                          pass_fds=(addr_w,),
+                          close_fds=True,
```

Comment:
> Providing any pass_fds forces close_fds to be True

From here: https://docs.python.org/3/library/subprocess.html

And that also answers my above question :) (but the read one is still open, isn't it?)

Reply:
right, I forgot to close that that after the read(). It pretty well doesn't matter, but for clarity/cleanliness I added a close for it as well. The subprocess only gets the write end, teh read end is closed automatically due to `closes_fds` indeed.